### PR TITLE
feat: Pulse Image

### PR DESCRIPTION
**Pull request by calebpanza**
## ✏️ Description
There is a new UI Kit component that we need to introduce in order to launch Daily Habits: PulseImage.

For reference, check out the [Figma Component](https://www.figma.com/file/OvEmzj97LwXzL0E5yBTvfB/pulse-image?node-id=4459%3A68805) and associated documentation. The pulse image can be used with 2 or 3 layers of an image. By default, there is a gradient overlay that is repeated on each layer, but you are free to change/negate this completely. It's preferred to use aspect ratios that are square or portrait. Landscape images are not advised. The default width is 58px, but that can be adjusted as need be.

## 🔬 To Test
1. Set the `PROJECT_VARIANT` to `storybook`
2. `yarn && yarn start` and `yarn ios`
3. Check out the `PulseImage` stories

## 📸 Screenshots
| Feature | Simulator | Figma | Links | Design Considerations |
| --- | --- | --- | --- | ---|
| Default Rendering | CleanShot 2022-11-08 at 15 05 23 | CleanShot 2022-11-08 at 15 17 04 | [Link to Figma](https://www.figma.com/file/OvEmzj97LwXzL0E5yBTvfB/pulse-image?node-id=4459%3A68805) | n/a |
| Displaying an Icon | CleanShot 2022-11-08 at 15 05 10 | CleanShot 2022-11-08 at 15 21 18 | n/a | n/a |
| Color Adjustments | CleanShot 2022-11-08 at 15 04 57 | CleanShot 2022-11-08 at 15 27 47 | n/a | n/a |

## ⚠️ To-do Before Merge

- [ ] [This branch](https://www.figma.com/file/YHJLj8pdFxWG9npF2YmB3r/branch/OvEmzj97LwXzL0E5yBTvfB/UI-Kit-2.0?node-id=2605%3A13981) in Figma needs to be merged into the main UI Kit (@conrad-vanl idk what the processor for reviews/approvals looks like for that)

## 📝 Checklist

- [x] I have performed a self-review of my code
- [x] I have tested my code works in the project (nothing breaks)
- [x] I have commented my code in hard-to-understand areas
- [x] New and existing tests and linting pass locally with my changes
- [x] I have made corresponding changes to the documentation (new envs?)

### CODE REVIEW

**Comment by conrad-vanl** [comment]
Looks good from a visual perspective (I did not look at the code). Figma branch can be merged.

File: templates/mobile/src/ui-kit/components/PulseImage/PulseImage.js [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+  numberOfLayers,
+  iconName,
+  iconSize,
+  iconWeight,
+  style,
+  overlayColors,
+  ...imageProps
+}) => {
+  const theme = useTheme();
+  const colors = overlayColors.map((colorKey) => get(theme.colors, colorKey));
```
Comment (jordanwade) [inline templates/mobile/src/ui-kit/components/PulseImage/PulseImage.js:40]: If you want to get rid of `lodash/get` I think you can use optional chaining here instead.

**Review by jordanwade — APPROVED** [review]

File: templates/mobile/src/ui-kit/components/PulseImage/PulseImage.styles.js [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+ * PulseImage.styles.js
+ *
+ * @description Styled Components for PulseImage
+ */
+
+import styled from 'styled-components/native';
+
+import { BlurView } from '@react-native-community/blur';
+import MaskedView from '@react-native-masked-view/masked-view';
+import { LinearGradient } from 'react-native-image-filter-kit';
```
[the remark this answers is not in the document]
Comment (calebpanza) [reply templates/mobile/src/ui-kit/components/PulseImage/PulseImage.styles.js:11]: Good catch! Removing now

File: templates/mobile/src/ui-kit/components/PulseImage/PulseImage.js (hunk 1 above)
Comment (calebpanza) [reply templates/mobile/src/ui-kit/components/PulseImage/PulseImage.js:40]: Color values are nested, so it's like `base.primary` or `text.secondary` so I don't think optional chaining will work?

**Pull request merged**